src/UnitOfWork.php

```diff
@@ -564,6 +571,29 @@ public function addManaged($entity)
         $this->manageEntityReference($oid);
     }
 
+    /**
+     * @param object $entity
+     *
+     * @return bool
+     */
+    public function isManaged($entity)
+    {
+        return isset($this->entityIds[spl_object_hash($entity)]);
+    }
+
+    private function removeManaged($entity)
```

Comment:
`detach` is recursive and uses `removeManged`. `detach` does also remove from the schedule arrays